infrastructure-examples/v2_inventory.yml.tftpl

```diff
@@ -0,0 +1,23 @@
+---
+all:
+  children:
+    benchmarkers:
+      hosts:
+%{ for name, values in servers.machines ~}
+%{   if values.tags.type == "dbt2-driver" ~}
+        ${ name }:
+          ansible_host: ${ values.public_ip }
+          private_ip: ${ values.private_ip }
+          ansible_user: ${ values.operating_system.ssh_user }
+%{   endif ~}
+%{ endfor ~}
+    targets:
```

Comment:
Where do we include information of primary or standbys something like given below:
```
---
all:
  children:
    pemserver:
      hosts:
        pemserver1:
          ansible_host: xxx.xxx.xxx.xxx
          private_ip: xxx.xxx.xxx.xxx
    barmanserver:
      hosts:
        barman1:
          ansible_host: xxx.xxx.xxx.xxx
          private_ip: xxx.xxx.xxx.xxx
    primary:
      hosts:
        primary1:
          ansible_host: xxx.xxx.xxx.xxx
          private_ip: xxx.xxx.xxx.xxx
          pem_agent: true
          pem_server_private_ip: xxx.xxx.xxx.xxx
          barman: true
          barman_server_private_ip: xxx.xxx.xxx.xxx
          barman_backup_method: rsync
    standby:
      hosts:
        standby1:
          ansible_host: xxx.xxx.xxx.xxx
          private_ip: xxx.xxx.xxx.xxx
          upstream_node_private_ip: xxx.xxx.xxx.xxx
          replication_type: synchronous
          pem_agent: true
          pem_server_private_ip: xxx.xxx.xxx.xxx
          barman: true
          barman_server_private_ip: xxx.xxx.xxx.xxx
          barman_backup_method: postgres
        standby2:
          ansible_host: xxx.xxx.xxx.xxx
          private_ip: xxx.xxx.xxx.xxx
          upstream_node_private_ip: xxx.xxx.xxx.xxx
          replication_type: asynchronous
          pem_agent: true
          pem_server_private_ip: xxx.xxx.xxx.xxx
          barman: true
          barman_server_private_ip: xxx.xxx.xxx.xxx
          barman_backup_method: postgres
    hammerdb:
      hosts:
        hammerdb1:
          ansible_host: xxx.xxx.xxx.xxx
          private_ip: xxx.xxx.xxx.xxx
```

Reply:
Taken from past postgres-deployment run with `EDB-RA-3`. The following uses jinja2 instead of terraform's templating but the concept is the similar. I do some processing in the template to order the machines based off their tags. Found here: https://github.com/EnterpriseDB/postgres-deployment/blob/master/edbdeploy/data/templates/inventory.yml.j2

```yaml
---
all:
  children:
    pemserver:
      hosts:
        pemserver1.new3aws.internal:
          ansible_host: 18.236.146.1
          private_ip: 10.2.1.85
    barmanserver:
      hosts:
        barmanserver1.new3aws.internal:
          ansible_host: 52.37.251.244
          private_ip: 10.2.1.130
          pem_agent: true
          pem_server_private_ip: 10.2.1.85
    primary:
      hosts:
        epas1.new3aws.internal:
          ansible_host: 52.12.169.50
          private_ip: 10.2.1.54
          barman: true
          barman_server_private_ip: 10.2.1.130
          barman_backup_method: postgres
          pem_agent: true
          pem_server_private_ip: 10.2.1.85
    standby:
      hosts:
        epas2.new3aws.internal:
          ansible_host: 52.34.244.33
          private_ip: 10.2.1.13
          barman: true
          barman_server_private_ip: 10.2.1.130
          barman_backup_method: postgres
          replication_type: synchronous
          upstream_node_private_ip: 10.2.1.54
          pem_agent: true
          pem_server_private_ip: 10.2.1.85
        epas3.new3aws.internal:
          ansible_host: 52.89.55.83
          private_ip: 10.2.1.124
          barman: true
          barman_server_private_ip: 10.2.1.130
          barman_backup_method: postgres
          replication_type: asynchronous
          upstream_node_private_ip: 10.2.1.54
          pem_agent: true
          pem_server_private_ip: 10.2.1.85
    pgpool2:
      hosts:
        pgpool21.new3aws.internal:
          ansible_host: 54.188.81.15
          private_ip: 10.2.1.87
          primary_private_ip: 10.2.1.54
          pem_agent: true
          pem_server_private_ip: 10.2.1.85
        pgpool22.new3aws.internal:
          ansible_host: 54.200.184.116
          private_ip: 10.2.1.179
          primary_private_ip: 10.2.1.54
          pem_agent: true
          pem_server_private_ip: 10.2.1.85
        pgpool23.new3aws.internal:
          ansible_host: 18.237.102.99
          private_ip: 10.2.1.94
          primary_private_ip: 10.2.1.54
          pem_agent: true
          pem_server_private_ip: 10.2.1.85
```

Machines from the infrastructure file. Note that the key names do not matter in the above output, it is all based off of tags set under each machine:
```yaml
machines:
  pem-server-0:
    instance_type: c5.xlarge
    volume:
      type: gp2
      size: 100
      iops: 250
      size_gb: 100
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: pem_server
      pg_type: epas
      index: 0
      count: 1
      priority: 0
    additional_volumes: []
  postgres-server-0:
    volume:
      type: gp2
      size: 50
      iops: 250
      size_gb: 50
    additional_volumes:
      - type: gp2
        size: 50
        iops: 250
        encrypted: false
        mount_point: /pgdata
        size_gb: 50
      - type: gp2
        size: 50
        iops: 250
        encrypted: false
        mount_point: /pgwal
        size_gb: 50
    instance_type: c5.2xlarge
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: postgres_server
      pg_type: epas
      index: 0
      count: 3
      priority: 0
      postgres_group: postgres_server
  postgres-server-1:
    volume:
      type: gp2
      size: 50
      iops: 250
      size_gb: 50
    additional_volumes:
      - type: gp2
        size: 50
        iops: 250
        encrypted: false
        mount_point: /pgdata
        size_gb: 50
      - type: gp2
        size: 50
        iops: 250
        encrypted: false
        mount_point: /pgwal
        size_gb: 50
    instance_type: c5.2xlarge
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: postgres_server
      pg_type: epas
      index: 1
      count: 3
      priority: 0
      postgres_group: postgres_server
      replication_type: synchronous
  postgres-server-2:
    volume:
      type: gp2
      size: 50
      iops: 250
      size_gb: 50
    additional_volumes:
      - type: gp2
        size: 50
        iops: 250
        encrypted: false
        mount_point: /pgdata
        size_gb: 50
      - type: gp2
        size: 50
        iops: 250
        encrypted: false
        mount_point: /pgwal
        size_gb: 50
    instance_type: c5.2xlarge
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: postgres_server
      pg_type: epas
      index: 2
      count: 3
      priority: 0
      postgres_group: postgres_server
      replication_type: asynchronous
  barman-server-0:
    instance_type: c5.2xlarge
    volume:
      type: gp2
      size: 50
      iops: 250
      size_gb: 50
    additional_volumes:
      - count: 1
        type: gp2
        size: 300
        iops: 250
        encrypted: false
        mount_point: /var/lib/barman
        size_gb: 300
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: barman_server
      pg_type: epas
      index: 0
      count: 1
      priority: 0
  pooler-server-0:
    instance_type: c5.xlarge
    volume:
      type: gp2
      size: 30
      iops: 250
      size_gb: 30
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: pooler_server
      pg_type: epas
      index: 0
      count: 3
      priority: 0
      pooler_type: pgpool2
      pooler_local: false
    additional_volumes: []
  pooler-server-1:
    instance_type: c5.xlarge
    volume:
      type: gp2
      size: 30
      iops: 250
      size_gb: 30
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: pooler_server
      pg_type: epas
      index: 1
      count: 3
      priority: 1
      pooler_type: pgpool2
      pooler_local: false
    additional_volumes: []
  pooler-server-2:
    instance_type: c5.xlarge
    volume:
      type: gp2
      size: 30
      iops: 250
      size_gb: 30
    image_name: image_default
    region: us-west-2
    zone_name: default
    tags:
      reference_architecture: edb-ra-3
      type: pooler_server
      pg_type: epas
      index: 2
      count: 3
      priority: 2
      pooler_type: pgpool2
      pooler_local: false
    additional_volumes: []
```